### ISSUE TITLE
Run fix command in worker

### ISSUE THREAD

**Issue by 5an1ty** [AtomLinter/linter-eslint#300]
When I run the command manually eslint fixes my file. But the built in command for atom doesn't work.

Example that works:
./node_modules/eslint/bin/eslint.js --fix ./subfolder/myfile.js

I'm not sure how to debug my atom application to find out why the command isn't working...

**Comment by steelbrain** [comment]
I just confirmed that it works for me, It removed the semi-colons when I used the fix command. It does have a bit lag though

**Comment by 5an1ty** [comment]
Not the case for me, doesn't work and doesn't throw an error. Any way I can provide you with more information?

Atom 1.2.1
OSX 10.11.1
linter eslint 5.2

.eslintrc:

    {
      "env": {
        "node": true,
        "browser": true
      },
      "plugins": [
        "eslint-plugin-react"
      ],
      "parser": "babel-eslint",
      "rules": {
        "no-unused-vars": 1,
        "camelcase": 1,
        "space-after-keywords": [2, "always"],
        "quotes": [2, "backtick", "avoid-escape"]
      },
      "extends": "eslint:recommended"
    }

**Comment by steelbrain** [comment]
@5an1ty I'll try to test it on OSX

**Comment by IanVS** [comment]
Not all situations can be auto-fixed by ESLint.  It might be helpful if you can give a reproducible test case.

**Comment by 5an1ty** [comment]
@IanVS I took the most straightforward code possible, just fixing quotes as you can see in my requirements:

    "quotes": [2, "backtick", "avoid-escape"]

A sufficient example would be:

    var test = 'testing';

**Comment by albertosantini** [comment]
Also in my Windows box I get the same issue.
No errors in the console, no exceptions, and no fix.

I did a few investigations.

I have been using `Use global ESLint installation` (and I installed all my plugins globally), but `Fix File` feature uses always embedded eslint.

If I uncheck that setting, I get an error, when I try to lint the file: `Error: Cannot find module 'eslint-config-eslint'`. This is due to that plugin is not installed (of course) in the embedded eslint.

Indeed reducing my config `.eslintrc`, without any plugin, the fix works nice:

```
rules:
    quotes: 2
```

See https://github.com/AtomLinter/linter-eslint/blob/master/lib/main.js#L79

- Catch eventually any error from stderr BufferedNodeProcess or an exit code
https://atom.io/docs/api/v1.2.1/BufferedNodeProcess#instance-constructor

- Use global eslint accordingly to the settings or find a way to use (global) plugins

**Comment by steelbrain** [comment]
I think the best way to solve this is to make the fix command use the worker, not only will it be faster but also more flexible

**Comment by albertosantini** [comment]
ESLint docs state the following sentence:
http://eslint.org/docs/user-guide/configuring#configuring-plugins

Note: A globally-installed instance of ESLint can only use globally-installed ESLint plugins. A locally-installed ESLint can make use of both locally- and globally- installed ESLint plugins.

Maybe it is an eslint bug. 
Indeed, when I try to lint the file from the command line, using a locally installed ESlint, I get the error: `Error: Cannot find module 'eslint-config-eslint'.` 

Maybe I cannot understand how ESLint works. Need to investigate.

**Comment by 5an1ty** [comment]
Just wanted to mention that my issue is completely unrelated to what @albertosantini is experiencing. I am using eslint and it's plugins locally.

**Comment by albertosantini** [comment]
@5an1ty What if you execute the following command in the folder where the file `foo.js`?

`node /your-path/.atom/packages/linter-eslint/node_modules/eslint/bin/eslint.js foo.js --fix`

**Comment by IanVS** [comment]
@albertosantini, in your example there, you are using the `eslint` that comes packaged with `linter-eslint`, which is neither a globally installed version nor the local version.

**Comment by albertosantini** [comment]
@IanVS That is my point: it is the command executed by `linter-eslint`.

See https://github.com/AtomLinter/linter-eslint/blob/master/lib/main.js#L79
Notice `__dirname`.

**Comment by IanVS** [comment]
Ah, I see.  It should only be executed by linter-eslint as a last resort, if you don't have a local `eslint` or a global `eslint`.  But when fixing, it always fails back to the installed version.

**Comment by 5an1ty** [comment]
@albertosantini then I get the following error:
    Error: Cannot find module 'eslint-plugin-react'

While running: 
    ./node_modules/eslint/bin/eslint.js --fix ./subfolder/myfile.js

does work...
